deeprank2/molstruct/residue.py

```diff
@@ -1,22 +1,25 @@
-from typing import Optional
+from typing import TYPE_CHECKING, Optional
 
 import numpy as np
+
 from deeprank2.molstruct.aminoacid import AminoAcid
 from deeprank2.molstruct.structure import Chain
 from deeprank2.utils.pssmdata import PssmRow
 
+if TYPE_CHECKING:
```

Comment:
Why do we need this if?

Reply:
This avoids circular imports. We cannot import `Residue` from atom.py _AND_ `Atom` from residue.py. Like this it only uses the import for type hint checking:

https://adamj.eu/tech/2021/05/13/python-type-hints-how-to-fix-circular-imports/